### PR TITLE
Add Functions slide and lab

### PR DESCRIPTION
**Pull request by juliaogris**

### CODE REVIEW

File: content/gotraining.slide [hunk 1]
```diff
@@ -488,15 +488,21 @@ As part of all labs
 
 * Functions
 
-- Functions
-- Multiple Return Values
-- Variadic Functions
-- Closures
+- [[https://goplay.space/#M_EBwjCDT77][Functions]]: parameters and return values
```
Comment (camscale) [inline content/gotraining.slide]: New link https://goplay.space/#gHOzTi05LkI

(s/gimmi/gimme/)

File: samples/functions2.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import "fmt"
+
+// Multiple parameters of same type shorthand
+// equivalent to `func sum(a int, b int) int`
+func sum(a, b int) int {
+	return a + b
+}
+
+// Multiple return values
+func gimmiTwo() (string, int) {
```
Comment (camscale) [inline samples/functions2.go]: s/gimmi/gimme/

fixed at https://goplay.space/#gHOzTi05LkI

File: samples/functions2.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	case "Melbourne":
+		lat, lng = -37.8136, 144.9631
+	}
+	return
+}
+
+func main() {
+	fmt.Println("sum(2,5)", sum(2, 5))
+	fmt.Println(location("Bangalore"))
+	fmt.Println(gimmiTwo())
```
Comment (camscale) [inline samples/functions2.go]: s/gimmi/gimme/

also fixed at https://goplay.space/#gHOzTi05LkI

File: samples/functions2.go [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+	return "0b10", 2
+}
+
+// Named return values, use with care
+func location(city string) (lat, lng float64) {
+	switch city {
+	case "Bangalore", "Bangaluru":
+		lat, lng = 12.9716, 77.5946
+	case "Melbourne":
+		lat, lng = -37.8136, 144.9631
```
Comment (camscale) [inline samples/functions2.go:23]: speaker notes: these are google lat/longs. bing gives slightly different results.

File: content/gotraining.slide [hunk 5]
```diff
@@ -488,15 +488,21 @@ As part of all labs
...
+- [[https://goplay.space/#Kc-7LOUq_Ls][Variadic Functions]]
```
Comment (camscale) [inline content/gotraining.slide]: new link: https://goplay.space/#-G75RWA1DW1

Remove parens from import

File: samples/variadic.go [hunk 6]
```diff
@@ -0,0 +1,25 @@
+// Variadic
+package main
+
+import (
+	"fmt"
+)
```
Comment (camscale) [inline samples/variadic.go]: Remove parens from single import for consistency with the rest of the samples.

new link: https://goplay.space/#-G75RWA1DW1

File: content/gotraining.slide [hunk 7]
```diff
@@ -488,15 +488,21 @@ As part of all labs
...
+- [[https://goplay.space/#olrpJoSte3n][Closures]]
 
 
-* Lab 4
+* Lab 4 - numeronym
 
-TODO
+- Create an executable go program in directory `3_letters`
```
Comment (camscale) [inline content/gotraining.slide]: s/3_letters/4_numeronym/

File: content/gotraining.slide [hunk 8]
```diff
@@ -488,15 +488,21 @@ As part of all labs
...
+- Write a function that returns a slice of numeronyms for its input strings
```
Comment (camscale) [inline content/gotraining.slide]: append colon

File: content/gotraining.slide [hunk 9]
```diff
@@ -488,15 +488,21 @@ As part of all labs
...
+
+	func numeronyms(vals ...string) []string
+
+- Call `fmt.Prinln(numeronyms("accessibility",` `"Kubernetes",` `"abc"))` in `main` to print
```
Comment (camscale) [inline content/gotraining.slide]: s/Prinln/Println/

append colon

File: content/gotraining.slide [hunk 10]
```diff
@@ -488,15 +488,21 @@ As part of all labs
...
+
+	[a11y K8s abc]
```
Comment (camscale) [inline content/gotraining.slide:505]: a1c? :)  what's the lowest number you put in - 2 I guess. perhaps the class can discuss this.

**Review by camscale — COMMENTED** [review]
some fixes...

File: samples/functions2.go (hunk 4 above)
Comment (juliaogris) [reply samples/functions2.go:23]: ACK

File: content/gotraining.slide (hunk 1 above)
Comment (juliaogris) [reply content/gotraining.slide]: Done.

File: samples/functions2.go (hunk 3 above)
Comment (juliaogris) [reply samples/functions2.go]: Done.

File: samples/functions2.go (hunk 2 above)
Comment (juliaogris) [reply samples/functions2.go]: Done.

File: content/gotraining.slide (hunk 5 above)
Comment (juliaogris) [reply content/gotraining.slide]: Done.

File: samples/variadic.go (hunk 6 above)
Comment (juliaogris) [reply samples/variadic.go]: Done.

File: content/gotraining.slide (hunk 7 above)
Comment (juliaogris) [reply content/gotraining.slide]: Done.

File: content/gotraining.slide (hunk 9 above)
Comment (juliaogris) [reply content/gotraining.slide]: Done.

File: content/gotraining.slide (hunk 10 above)
Comment (juliaogris) [reply content/gotraining.slide:505]: yeah, good idea to discuss.... i'd say it should shorten the input....

File: samples/functions2.go [hunk 11]
```diff
@@ -28,5 +28,5 @@ func location(city string) (lat, lng float64) {
 func main() {
 	fmt.Println("sum(2,5)", sum(2, 5))
 	fmt.Println(location("Bangalore"))
-	fmt.Println(gimmiTwo())
+	fmt.Println(gimmeTwo())
```
Comment (camscale) [inline samples/functions2.go:31]: speaker notes: issues with nesting function calls where the inner returns multiple values. you can't pass any other values.

File: content/gotraining.slide (hunk 10 above)
Comment (camscale) [reply content/gotraining.slide:505]: definitely should shorted, but a minimum amount? you probably wouldn't want to do d2e for dude, for example. but they can try coding up a solution where a var or constant controls it, for a bit of extra work.

**Review by camscale — APPROVED** [review]
LGTM

**Pull request merged**